test/boost/tablets_test.cc

```diff
@@ -2098,6 +2098,15 @@ struct calculate_tablet_replicas_for_new_rf_config
     service::tablet_replica_calculation_status expected_reallocation_status;
 };
 
+__attribute__((no_sanitize("undefined")))
+int64_t double_to_int64_t(double d) {
+    // Double to unsigned long conversion will overflows if the
+    // input is greater than numeric_limits<long>::max(), so divide by two and
+    // multiply again later.
+    auto scale = std::numeric_limits<unsigned long>::max();
+    return static_cast<unsigned long>(d * static_cast<double>(scale >> 1)) << 1;
+};
+
```

Comment:
no motivation or context for the change in the patch log.
any why is it needed in this series?

Reply:
OK, I squashed this patch with the commit that introduces the tests.
I wanted to keep this change in a separate commit to make it apparent that I am suppressing an error, but I changed that to an explicit message in the commit.

Reply:
I wonder why it's not needed in network_topology_strategy_test...

Reply:
> I wonder why it's not needed in network_topology_strategy_test...

Yeah, interesting. Either way, I removed `double_to_int64_t` function from `tablets_test.cc`, as it was just a copy-paste of `d2t` function defined in `network_topology_strategy_test.cc`. It's still marked with a suppress comment, though.